app/src/components/map/DatasetPopup.tsx

```diff
@@ -0,0 +1,126 @@
+import Box from '@material-ui/core/Box';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Collapse from '@material-ui/core/Collapse';
+import Link from '@material-ui/core/Link';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import { SPATIAL_COMPONENT_TYPE } from 'constants/spatial';
+import { Feature } from 'geojson';
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import useDataLoader from 'hooks/useDataLoader';
+import React from 'react';
+import { useHistory } from 'react-router';
+
+export type BoundaryCentroidFeature = Feature & { properties: BoundaryCentroidFeatureProperties };
+
+export type BoundaryCentroidFeatureProperties = {
+  type: SPATIAL_COMPONENT_TYPE.BOUNDARY_CENTROID;
+  datasetTitle: string;
+  datasetID: string;
+};
+
+const useStyles = makeStyles(() => ({
+  modalContent: {
+    position: 'relative',
+    width: 300,
+    minHeight: 36
+  },
+  loading: {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)'
+  },
+  pointType: {
+    lineHeight: 'unset'
+  },
+  date: {
+    margin: 0,
+    lineHeight: 'unset'
+  },
+  table: {
+    marginTop: 16
+  },
+  tableCell: {
+    padding: 8,
+    fontSize: '0.625rem'
+  }
+}));
+
+const DatasetPopup: React.FC<React.PropsWithChildren<{ submissionSpatialComponentIds: number[] }>> = ({
+  submissionSpatialComponentIds
+}) => {
+  const classes = useStyles();
+  const api = useBiohubApi();
+  const history = useHistory();
+
+  const dataLoader = useDataLoader(() => {
+    return api.observation.getSpatialMetadata(submissionSpatialComponentIds);
+  });
+
+  dataLoader.load();
+
+  const { isLoading, isReady } = dataLoader;
+  const data = (dataLoader.data || []) as BoundaryCentroidFeatureProperties[];
+
+  const ModalContentWrapper: React.FC<any> = ({ children }) => <div className={classes.modalContent}>{children}</div>;
+
+  const MetadataHeader: React.FC<React.PropsWithChildren<{ title: string }>> = (props) => (
+    <Box mb={1}>
+      <Typography variant="overline" className={classes.pointType}>
+        Dataset
+      </Typography>
+      {props.title && (
+        <Typography className={classes.date} component="h6" variant="subtitle1">
+          {props.title}
+        </Typography>
+      )}
+    </Box>
+  );
+
+  const NoMetadataAvailable: React.FC = () => (
+    <Typography className={classes.date} component="h6" variant="body1">
+      No metadata available.
+    </Typography>
+  );
+
+  if (isLoading) {
+    return (
+      <ModalContentWrapper>
+        <div className={classes.loading}>
+          <CircularProgress size={24} color="primary" />
+        </div>
+      </ModalContentWrapper>
+    );
+  }
+
+  if (!data) {
```

Comment:
Note that `Boolean([])` resolves to `true`; If `const data = (dataLoader.data || [])` then we should never expect this conditional to be true. Should this be `if (!data.length)`?